core/src/impl/Kokkos_GraphImpl_fwd.hpp

```diff
@@ -44,10 +44,20 @@ struct is_graph_capture : public std::false_type {};
 template <typename T>
 inline constexpr bool is_graph_capture_v = is_graph_capture<T>::value;
 
+template <typename ExecutionSpace, typename Functor>
+struct GraphNodeThenHostImpl;
+
+template <typename T, class Enable = void>
+struct is_graph_host : public std::false_type {};
```

Comment:
Probably worth thinking whether we want to have `then_host` systematically in the names.